File: Module06/bank-account-comparison/bank_account.py
class BankAccount:
    def __init__(self, account_number: int, owner: str):
        self.__account_number = account_number
        self._owner = owner
        self._balance: float = 0

    @property
    def account_number(self) -> int:
        return self.__account_number

    @property
    def owner(self) -> str:
        return self._owner

    @owner.setter
    def owner(self, owner: str) -> None:
        if not isinstance(owner, str) or len(owner) < 3:
            raise ValueError("Owner must be a string with a minimum length of 3")
        self._owner = owner

    @property
    def balance(self):
        return self._balance

    def get_balance(self) -> float:
        return self._balance

    def deposit(self, amount: float) -> None:
        self._balance += amount

    def withdraw(self, amount: float) -> None:
        self._balance -= amount

    def __str__(self) -> str:
        return (
            f"Account number {self.__account_number} "
            f"owned by {self._owner} "
            f"with balance {self._balance}"
        )

File: Module07/bank-account-comparison/BankAccount.java
package edu.au;

public class BankAccount {

	private Person owner;
	private int account_number;
	private double balance;
	
	public BankAccount(int account_number, String owner) {
		this.account_number = account_number;
		this.owner = owner;
		this.balance = 0;
	}
	
	
	public String getOwner() {
		return owner;
	}

	public void setOwner(String owner) {
		if (owner == null || owner.length() < 3)
			throw new IllegalArgumentException("owner must be non-null and at least 3 characters");
		this.owner = owner;
	}
		
	public double getBalance() {
		return balance;
	}
	
	public int getAccountNumber() {
		return account_number;
	}
	
	public void deposit(double amount) {
		balance += amount;
	}
	
	public void withdraw(double amount) {
		balance -= amount;
	}


	public String toString() {
		return "Account number: " + account_number +
                    "  owner: " + owner +
                    "  with balance: " + balance;
	}
}

